mlrun/runtimes/base.py

```diff
@@ -403,7 +404,13 @@ def _generate_runtime_env(self, runobj: RunObject = None) -> Dict:
             "MLRUN_DEFAULT_PROJECT": self.metadata.project or config.default_project
         }
         if runobj:
-            runtime_env["MLRUN_EXEC_CONFIG"] = runobj.to_json()
+            # Since the `params` attribute within each notification object can be large,
+            # it has the potential to cause errors and is unnecessary for the notification functionality.
+            # Therefore, in this section, we remove the `params` attribute from each notification object.
+            runobj_for_exec_config = deepcopy(runobj)
+            for notification in runobj_for_exec_config.spec.notifications:
+                notification.params = {}
+            runtime_env["MLRUN_EXEC_CONFIG"] = runobj_for_exec_config.to_json()
```

Comment:
not all object properties can be coppied. nonetheless, if body is too large, you now make it x2 at the time of running it.
I suggest to `runobj.to_json()` it, then pop the keys out and shove it to `MLRUN_EXEC_CONFIG`.
also, note that  `to_env` function (under run template) does the exact same things. so mind the code dup of your solution.

Reply:
also, add ut to cover such case